src/services/dashboards/modules/dashboard-toolset/DashboardToolset.vue

```diff
@@ -1,16 +1,48 @@
 <template>
     <div class="dashboard-toolset">
-        <dashboard-date-range-badge />
-        <dashboard-date-dropdown />
+        <dashboard-date-range-badge :date-range="state.proxyDateRange" />
+        <dashboard-date-dropdown :date-range.sync="state.proxyDateRange" />
         <currency-select-dropdown default-mode />
     </div>
 </template>
 
-<script setup lang="ts">
+<script lang="ts">
+import type { PropType } from 'vue';
+import { reactive } from 'vue';
+
+import { useProxyValue } from '@/common/composables/proxy-state';
 import CurrencySelectDropdown from '@/common/modules/dropdown/currency-select-dropdown/CurrencySelectDropdown.vue';
 
+import type { DateRange } from '@/services/dashboards/config';
 import DashboardDateDropdown from '@/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue';
 import DashboardDateRangeBadge from '@/services/dashboards/modules/dashboard-toolset/DashboardDateRangeBadge.vue';
+
+export default {
+    name: 'DashboardToolset',
+    components: {
+        CurrencySelectDropdown,
+        DashboardDateDropdown,
+        DashboardDateRangeBadge,
+    },
+    props: {
+        dateRange: {
+            type: Object as PropType<DateRange>,
+            default: undefined,
+            required: true,
+        },
+    },
+    setup(props, { emit }) {
+        const state = reactive({
+            proxyDateRange: useProxyValue('dateRange', props, emit),
+        });
+
+        return {
+            state,
+        };
+    },
+};
+
+
```

Comment:
I also had the same experience. I couldn't use emit in script setup..

Reply:
So sad 😂😂😂